### PR TITLE
Feat/subdomains

### PR DESCRIPTION
**Pull request by Th0rgal**
This fixes the fact you couldn't reuse a subdomain after resetting all subdomains of a specific domain

### CODE REVIEW

File: src/naming/main.cairo [hunk 1]
```diff
@@ -107,6 +105,14 @@ func domain_to_expiry{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_chec
     return (domain_data.expiry,);
 }
 
+@view
+func domain_to_data{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
```
Comment (0xBenaparte) [inline src/naming/main.cairo:109]: This is cool and useful 👍

File: tests/integration/test_reset_subdomains.cairo [hunk 2]
```diff
@@ -62,3 +79,50 @@ func test_reset_subdomains{syscall_ptr: felt*, range_check_ptr, pedersen_ptr: Ha
...
+        warp(1, context.naming_contract)
+    %}
+
+    let token_id = 1;
+    StarknetId.mint(starknet_id_contract, token_id);
+
+    let token_id2 = 2;
+    StarknetId.mint(starknet_id_contract, token_id2);
+
+    // should mint a domain and create a subdomain
```
Comment (0xBenaparte) [inline tests/integration/test_reset_subdomains.cairo:102]: You only mint a domain, you don't create a subdomain here do you ?

File: tests/integration/test_reset_subdomains.cairo [hunk 3]
```diff
@@ -62,3 +79,50 @@ func test_reset_subdomains{syscall_ptr: felt*, range_check_ptr, pedersen_ptr: Ha
...
+    assert token_id = token_id2;
+
+    // should reset the subdomain owner
+    Naming.reset_subdomains(naming_contract, 1, new ('alpha'));
+    let (token_id) = Naming.domain_to_token_id(naming_contract, 2, new ('bravo', 'alpha'));
+    assert token_id = 0;
+
+    let token_id3 = 3;
+    StarknetId.mint(starknet_id_contract, token_id3);
+    // should mint the subdomain again
```
Comment (0xBenaparte) [inline tests/integration/test_reset_subdomains.cairo:117]: You have to place this before the declaration of token_id3 in order for it to be clearer ;)

File: tests/integration/test_reset_subdomains.cairo [hunk 4]
```diff
@@ -62,3 +79,50 @@ func test_reset_subdomains{syscall_ptr: felt*, range_check_ptr, pedersen_ptr: Ha
...
+    Naming.transfer_domain(naming_contract, 2, new ('bravo', 'alpha'), token_id3);
+    let (token_id) = Naming.domain_to_token_id(naming_contract, 2, new ('bravo', 'alpha'));
+    assert token_id = token_id3;
```
Comment (0xBenaparte) [inline tests/integration/test_reset_subdomains.cairo:120]: Nice test gg

**Review by 0xBenaparte — CHANGES_REQUESTED** [review]
Just a few changes/questions on the comments of the tests. 

Test coverage is increasing ;)

**Comment by fricoben** [comment]
I marked this PR as 1 day in Only Dust but @Th0rgal told me that this PR and [this PR ](https://github.com/starknet-id/contract/pull/8) are one day (0.5 this one and 0.5 the other one). 

So consequently I'll not report this PR on [Only Dust](https://github.com/starknet-id/contract/pull/8)

**Review by DeepMintingValue — APPROVED** [review]

**Review by fricoben — APPROVED** [review]

**Pull request merged**